components/form-builder/hooks/useAllowPublish.tsx

```diff
@@ -1,32 +1,52 @@
 import { useCallback } from "react";
-import { FormSchema, publishRequiredFields } from "../types";
+import { Description, ElementType, FormSchema, publishRequiredFields, Title } from "../types";
 import useTemplateStore from "../store/useTemplateStore";
 
 const MissingTranslation = {};
 
-const isTranslated = (elementEn: string, elementFr: string) => {
-  if (elementEn || elementFr) {
-    if (!elementEn || !elementFr) {
-      throw MissingTranslation;
+const isTitleTranslated = (element: Title) => {
+  if (!element.titleEn || !element.titleFr) {
+    throw MissingTranslation;
+  }
+};
+
+const isDescriptionTranslated = (element: Description) => {
+  if (!element.descriptionEn || !element.descriptionFr) {
+    throw MissingTranslation;
+  }
+};
+
+const isElementTranslated = (element: ElementType) => {
+  if (element.type === "richText") {
+    isDescriptionTranslated(element.properties);
```

Comment:
Can we do an early return here to get rid of the else?

Reply:
refactored a bit, think it's better